submissions/Siusarna/data-structures/data-structures/linkedList.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable no-underscore-dangle */
+
+const Node = require('./Element.js');
+
+function getSuccessorNode(prevNode, successor) {
+  let successorNode = prevNode;
+  if (!successor) {
+    return successorNode;
+  }
+  // successor value saved in prevNode or prevNode._next() so we check these node
+  if (prevNode.getValue() === successor) {
+    successorNode = prevNode;
+  } else if (prevNode.getNext()) {
+    successorNode = prevNode.getNext();
+  } else {
+    throw new Error('not found this successor');
+  }
+  return successorNode;
+}
+
+module.exports = class LinkedList {
+  constructor() {
+    this._head = null;
+    this._size = 0;
+  }
+
+  searchForPrevNode(inputValue) {
+    let iteratorNode = this._head;
+    while (
+      iteratorNode.getNext() &&
+      iteratorNode.getNext().getValue() !== inputValue
+    ) {
+      iteratorNode = iteratorNode.getNext();
+    }
+    return iteratorNode;
+  }
+
+  insert(inputValue, successor = null) {
+    if (this._head === null) {
+      this._head = new Node(inputValue);
+      this._size += 1;
+    } else {
+      const prevNode = this.searchForPrevNode(successor);
+      const successorNode = getSuccessorNode(prevNode, successor);
```

Comment:
If I understood you, then I don't add function `getSuccessorNode` in class becouse I write this function special for insert method and don't image how can use this function on the other.

Reply:
If I understood you, I'm not adding the `getSuccessorNode` function to the class because I'm writing this function specifically for the insert method, and I can't imagine how it can be used otherwise.
We can use the `searchForPrevNode` function for work with list, so I'm adding this to the class, but `getSuccessorNode` is only for the` insert 'method

Reply:
I see. But I am a little bit confused.

> function to the class because I'm writing this function specifically for the insert method, and I can't imagine how it can be used otherwise.

So the only user of this function, is a class `LinkedList` and this function can't be reused outside of it? Same situation as with `searchForPrevNode`, no?

Reply:
I think if a user imports this class, they can use `searchForPrevNode` for their tasks with list, such as `"swap two nodes"` (I had a similar task at university), but `getSuccessorNode` I think the user can't use this function for outside.
If you recommend adding this method to the class, I'll do it

Reply:
So this is because of privacy of method? I see. Have you tried [private fields](https://thecodebarbarian.com/nodejs-12-private-class-fields.html)?

> If you recommend adding this method to the class, I'll do it

Actually, it is more than one way how you can solve such things:

1. Just add method to a class and then anyone can use it
2. Add underscore to name of method to show it's privacy and hope, that no one will use it
3. Create function as you did
4. Or don't use class at all...

In the end, as I told, I was just interested in reasoning.

Reply:
In order to be confident that no one will use it, I just choose the variant with function)

Reply:
But if you will use private field no one will use it anyway :)